src/traces/sankey/plot.js

```diff
@@ -0,0 +1,258 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var render = require('./render');
+var Fx = require('../../plots/cartesian/graph_interact');
+var d3 = require('d3');
+var Color = require('../../components/color');
+
+function renderableValuePresent(d) {return d !== '';}
+
+function ownTrace(selection, d) {
+    return selection.filter(function(s) {return s.key === d.traceId;});
+}
+
+function makeTranslucent(element, alpha) {
+    d3.select(element)
+        .select('path')
+        .style('fill-opacity', alpha);
+}
+
+function makeTextContrasty(element, hue) {
+    d3.select(element)
+        .select('text.name')
+        .style('fill', Color.contrast(hue, 0, 40));
+}
+
+function relatedLinks(d) {
+    return function(l) {
+        return d.node.sourceLinks.indexOf(l.link) !== -1 ||  d.node.targetLinks.indexOf(l.link) !== -1;
+    };
+}
+
+function relatedNodes(l) {
+    return function(d) {
+        return d.node.sourceLinks.indexOf(l.link) !== -1 ||  d.node.targetLinks.indexOf(l.link) !== -1;
+    };
+}
+
+function nodeHoveredStyle(sankeyNode, d, sankey) {
+
+    sankeyNode.select('.nodeRect').style('stroke-width', 1);
+
+    if(d && sankey) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyLink')
+            .filter(relatedLinks(d))
+            .call(linkHoveredStyle);
+    }
+}
+
+function nodeNonHoveredStyle(sankeyNode, d, sankey) {
+
+    sankeyNode.select('.nodeRect').style('stroke-width', 0.5);
+
+    if(d && sankey) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyLink')
+            .filter(relatedLinks(d))
+            .call(linkNonHoveredStyle);
+    }
+}
+
+function linkHoveredStyle(sankeyLink, d, sankey) {
+
+    sankeyLink.style('stroke-opacity', 0.4);
+
+    if(d && sankey) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyNode')
+            .filter(relatedNodes(d))
+            .call(nodeHoveredStyle);
+    }
+}
+
+function linkNonHoveredStyle(sankeyLink, d, sankey) {
+
+    sankeyLink.style('stroke-opacity', function(d) {return d.tinyColorAlpha;});
+
+    if(d && sankey) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyNode')
+            .filter(relatedNodes(d))
+            .call(nodeNonHoveredStyle);
+    }
+}
+
+var log = false
```

Comment:
Correct, this and some other things such as `index-sankey` are only for current dev. Will remove as it nears merging; now just helpful w/ churning out codepen with small bundle size.